### PR TITLE
extended examples and added developer advice section

### PR DESCRIPTION
**Pull request by stevefaulkner**
note:
'taco-button' tag name borrowed from Marcy Sutton
'tequila-button' tag name borrowed from Leonie Watson

### CODE REVIEW

**Comment by dglazkov** [comment]
Can you rebase this on top of the latest code? Something's off here.

**Comment by stevefaulkner** [comment]
> Can you rebase this on top of the latest code? Something's off here.
> my git skills are less than good, I will try again to sort it out my end though

**Comment by stevefaulkner** [comment]
@dglazkov don't know how but i think i sorted the issue and added self ack

File: spec/custom/index.html [hunk 1]
```diff
@@ -89,8 +89,9 @@ <h2 id="toc">Table of Contents</h2>
     <li><a href="#es6">Custom Elements and ECMAScript 6</a></li>
     <li><a href="#semantics">Custom Element Semantics</a>
     <ol>
-        <li><a href="#custom-tag-semantics">Custom Tag Semantics</a></li>
-        <li><a href="#type-extension-semantics">Type Extension Semantics</a></li>
+     <li><a href="#custom-tag-semantics">Custom Tag Semantics</a></li>
```
Comment (dglazkov) [inline spec/custom/index.html]: Would love to keep indentation here.

File: spec/custom/index.html [hunk 2]
```diff
@@ -764,52 +765,74 @@ <h2 id="es6">Custom Elements and ECMAScript 6</h2>
...
   </ul>
 <h3 id="custom-tag-semantics">Custom Tag Semantics</h3>
 <p>By default a <a href="#dfn-custom-tag">custom tag</a> has no special meaning at all. It <a href="http://www.w3.org/TR/html/dom.html#represents">represents</a> its children.</p>
-<p>For example a custom tag, could be named <em>taco-button</em> but it does not express the semantics of a HTML <a href="http://www.w3.org/TR/html/forms.html#the-button-element"><code>button</code></a> element simply due to its name. As instaniated a custom tag conveys a similar amount of semantics as a HTML <a href="http://www.w3.org/TR/html/grouping-content.html#the-div-element"><code>div</code></a> or <code><a href="http://www.w3.org/TR/html/text-level-semantics.html#the-span-element">span</a></code> element. The addition of visual styling and scripted events to the <em>taco-button</em> could provide hints as to the semantics and expected interaction behaviours of the custom element for <em>some</em> users, but for the semantics to be formally expressed, developers must convey the role, states and properties of the element to browser APIs.</p>
+<h4 id="custom-tag-example">Custom Tag Example</h4>
```
Comment (dglazkov) [inline spec/custom/index.html:781]: Pls add this to the TOC.

Comment (stevefaulkner) [inline spec/custom/index.html:781]: done

**Pull request merged**